src/main.py

```diff
@@ -1924,40 +1954,42 @@ def _check_primary_switchover(self, db_state, zk_state):
         if switchover_candidate is None:
             return None
 
-        if not self._candidate_is_sync_with_primary(db_state, switchover_candidate):
+        if not self._candidate_is_sync_with_primary(db_state.get('replics_info', []), switchover_candidate):
             return None
 
         logging.info('Scheduled switchover checks passed OK.')
         return switchover_candidate
 
-    def _all_side_replicas_turned_to_the_candidate(self, switchover_candidate):
-        candidate_app_name = helpers.app_name_from_fqdn(switchover_candidate)
-        replics_info = self.db.get_replics_info('primary')
-        app_names = [r['application_name'] for r in replics_info]
-        logging.info('Replica app names: %s', app_names)
-        return len(app_names) == 1 and candidate_app_name == app_names[0]
+    def _all_side_replicas_turned_to_the_candidate(self, old_primary, switchover_candidate, alive_ha_replics):
+        alive_ha_replics_names = set(helpers.app_name_from_fqdn(h) for h in alive_ha_replics if h not in (old_primary, switchover_candidate))
+        replics_info = self.db.get_replics_info('replica')
+        turned_replicas_names = set(r['application_name'] for r in replics_info if r['application_name']) & alive_ha_replics_names
+        logging.info('Replicas streaming from the candidate: %s, waiting for %s', turned_replicas_names, alive_ha_replics_names-turned_replicas_names)
+        return turned_replicas_names == alive_ha_replics_names
```

Comment:
Are we really required to wait for ALL replicas to turn?
I think it is a bit more complicated: we should definitely wait for all until timeout, but should not fail if after timeout at least quorum of replicas have turned.

I guess we should introduce function `_quorum_of_side_replicas_turned_to_the_candidate` and use it to check in case when https://github.com/yandex/pgconsul/pull/108/changes#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312R873 check fails.
Or change this function to count number of turned replicas and compare it to alive replicas count and quorum count.